lib/1.4/dml-builtins.dml

```diff
@@ -568,21 +581,76 @@ session dml_qname_cache_t _qname_cache;
 
 // Shared parameters for attributes, registers and connects, which all produce
 // attributes. Methods get_attribute and set_attribute are abstract.
-template _conf_attribute {
-    is object;
-
-    param configuration             default "optional";
-    param persistent                default false;
-    param internal                  default (!defined documentation
-                                             && !defined desc);
-
-    param readable default this.configuration != "none";
-    param writable default this.configuration != "none";
+template _conf_attribute is object {
+    shared method set_attribute(attr_value_t value) -> (set_error_t);
+    shared method get_attribute() -> (attr_value_t);
+
+    param _attr_name : const char *;
+    param _attr_name auto;
+    param _parent_obj_class : conf_class_t **;
+    param _parent_obj_class auto;
+    param _parent_obj_proxy_info : _attr_parent_obj_proxy_info_t;
+    param _parent_obj_proxy_info auto;
```

Comment:
I like your approach here. If I'd have written it I would probably just have given up on proxy attrs and kept them fully hard-coded; you manage to keep them soft-coded without sacrificing performance OR readability. Nice!

Reply:
:)